src/main/java/ui/Ui.java

```diff
@@ -0,0 +1,111 @@
+package ui;
+
+import exceptions.AliceException;
+import parser.Parser;
+import storage.TaskList;
+import tasks.Task;
+
+/**
+ * Responsible for printing output to the terminal, and reading input from the terminal.
+ * Encapsulates a parser and a scanner for reading user input.
+ */
+public class Ui {
+    private final Parser parser;
+
+    /**
+     * Initialises a Ui object.
+     *
+     * @param list the list to start the parser.
+     */
+    public Ui(TaskList list) {
+        parser = new Parser(list, this);
+    }
+
+    /**
+     * Returns the welcome message.
+     */
+    public String showWelcomeMessage() {
+        return "Hello! I am Alice!\nWhat can I do for you?";
+    }
+
+    /**
+     * Returns the exit message
+     */
+    public String showExitMessage() {
+        return "Bye. Hope to see you again soon!";
+    }
+
+    /**
+     * Returns the response of Alice.
+     *
+     * @param input the user's input.
+     * @return the response.
+     */
+    public String handleUserInput(String input) {
+        try {
+            return parser.parse(input);
+        } catch (AliceException e) {
+            return e.toString();
+        }
+    }
+
+    /**
+     * Returns the custom message when task is marked or unmarked.
+     *
+     * @param task the task which was handled.
+     * @param command "mark" or "unmark".
+     * @return the custom message.
+     */
+    public String getHandleTaskMessage(Task task, String command) {
+        if (command.equals("mark")) {
+            return "Noted! I've marked this task as done: \n" + task;
+        } else {
+            return "Ok, I've marked this task as not done yet: \n" + task;
+        }
+    }
+
+    /**
+     * Returns the custom message when task is deleted or added.
+     *
+     * @param task the task that was handled.
+     * @param command "delete" or "todo" or "deadline" or "event".
+     * @param size number of tasks in the list.
+     * @return the custom message.
+     */
+    public String getHandleTaskMessage(Task task, String command, int size) {
+        StringBuilder s = new StringBuilder();
+        if (command.equals("delete")) {
+            s.append("Noted. I've removed this task: \n");
+        } else {
+            s.append("Got it. I've added this task: \n");
+        }
+        s.append(task).append("\n");
+        s.append("Now you have ").append(size).append(" tasks in the list");
+        return s.toString();
+    }
+
+    /**
+     * Returns a string of all the tasks in the list.
+     *
+     * @param tasks the string representation of all the tasks.
+     * @param size number of tasks in the list.
+     * @return tasks in the list.
+     */
+    public String getListedTasks(String tasks, int size) {
```

Comment:
Maybe for this method, you can follow the naming for the other methods in the class (ending with the -Message suffix), I think this would be helpful for others to understand and to being consistent!